feat(shared-data, api): Add evo tips definition and restrictions on loading

Laura-Danielle opened the pull request:
# Overview

Add evo tips definition but as a labware so that liquid handling actions can happen in the 'labware'.

## Test Plan and Hands on Testing

Need to check that the stackup works as expected

## Changelog

- Add evo tips definition
- Add stacking component to nest deepwell plate
- Add error check for the evo tips definition so that it is not accidentally loaded onto the deck

## Review requests

Check that the labware role makes sense and the stacking part.

## Risk assessment

Low. Adding a new labware.

Laura-Danielle committed:
feat(shared-data, api): Add evo tips definition and restrictions on loading

Laura-Danielle committed:
feat(shared-data): Add evo flex tiprack adapter (#16707)

Laura-Danielle committed:
update definitions and fix typing issue

sfoster1 commented on api/src/opentrons/protocol_api/validation.py:
```diff
@@ -363,6 +372,27 @@ def ensure_definition_is_labware(definition: LabwareDefinition) -> None:
         )
 
 
+def ensure_labware_is_loadable(
```
This wants to be in the engine I reckon

sfoster1 reviewed (changes requested):
The verification logic wants to be in the engine but other than that looks good.

Laura-Danielle committed:
fix: ensure wells are the same height and the schema has the new labware role

Laura-Danielle committed:
remove validation, add some additional quirks

Laura-Danielle committed:
add additonal role for evo tips labware

Laura-Danielle committed:
add labware defs to new def list and modify validation

Laura-Danielle committed:
update stacking only command

jerader committed:
fix(protocol-designer): editing labware on top of a module and adapter fixes (#17161)

address RQA-3777 RQA-3804

koji committed:
chore(shared-data): remove NODE_OPTIONS=--openssl-legacy-provider (#17183)

* chore(shared-data): remove NODE_OPTIONS=--openssl-legacy-provider

jerader committed:
fix(protocol-designer): allow minutesSeconds timer fields to add more… (#17185)

… than 99 minutes

closes RQA-3817

y3rsh committed:
test(analyses): add 8.2 features to analyses battery (#17134)

jerader committed:
fix(step-generation): configureForVolume volume correctly set to am… (#17189)

…ount

closes RESC-366

Previously, the configure for volume volume was set to the transfer's total volume and not the chunk's volume. But this pr changed it so the volume was set to the chunk number! This pr fixes it so the volume is set to the actual volume

ahiuchingau committed:
feat(shared-data): add flex stacker definition and update module schema (#17196)

This PR adds a new module type for the Flex Stacker in shared-data,
module definition with some stubbed data and update the module schema so
we can use reference this in other part of the code base.

SyntaxColoring committed:
refactor(api): Delete obsolete re-exports (#17172)

caila-marashaj committed:
feat(api): inner well geometry unit tests (#17082)

sfoster1 committed:
chore: fix canbus test warnings (#17192)

Removes 40 or so warnings from tests on edge. Should be no functional
change.

sfoster1 reviewed (approved):
Excellent, looks good to me once checks pass!

sfoster1 committed:
chore: set abr-testing pydantic to 2.9 explicitly (#17200)

Matches the robot-side version.

koji committed:
fix(protocol-designer): add align-items to title part (#17165)

* fix(protocol-designer): add align-items to title part

koji committed:
chore(protocol-designer): remove * as React from react (#17168)

* chore(protocol-designer): remove * as React from react

koji committed:
fix(protocol-designer): fix paddings in release note modal (#17175)

* fix(protocol-designer): fix paddings in release note modal

jerader committed:
fix(protocol-designer): allow custom labware on adapters (#17162)

closes RQA-3779

jerader committed:
fix(protocol-designer): moving labware into waste chute step summary … (#17194)

…copy fix

closes RQA-3769

koji committed:
fix(protocol-designer): remove unused packages from pd (#17166)

* fix(protocol-designer): remove unused packages from pd

jerader committed:
fix(protocol-designer): release notes banner dismisses upon clicking x (#17199)

closes RQA-3821

koji committed:
fix(components, protocol-designer, app) fix some broken stories (#17208)

* fix(components, protocol-designer, app) fix some broken stories

koji committed:
fix(protocol-designer): fix small condition issues (#17182)

* fix(protocol-designer): fix small condition issues

jerader committed:
fix(protocol-designer): tooltip under path animation copy update (#17209)

closes RQA-3806

jerader committed:
refactor(protocol-designer): liquid class ff reducer typo fix (#17211)

TamarZanzouri committed:
chore(api): add absorbance reader module state update test (#17205)

koji committed:
chore: update vite version to 5.4.11 (#17075)

* chore: update vite version to 5.4.11

jerader committed:
fix(step-generation): only aspirate disposal volume in first chunk (#17202)

closes RQA-3807

koji committed:
refactor(components): refactor modal for web (app and pd) (#17193)

* refactor(components): refactor modal for web (app and pd)

Laura-Danielle committed:
modify the stacking depth for evotips and nest deepwell plate

ahiuchingau committed:
feat(shared-data): add flex stacker addressable area to deck definition (#17204)

caila-marashaj committed:
fix(shared-data): find latest definition if version not specified  (#17223)

koji committed:
fix(protocol-designer): unify time format to display the time (#17221)

* fix(protocol-designer): unify time format to display the time

caila-marashaj committed:
Revert "fix(shared-data): find latest definition if version not specified " (#17225)

jerader committed:
fix(protocol-designer): padding adjustments with forms (#17227)

closes RQA-3700 RQA-3693

ncdiehl11 committed:
fix(protocol-designer): fix small DQA bugs (#17228)

Fixes 3 small style bugs for 1) copy for importing a protocol on PD
landing page, 2) background color of selected off deck location, and 3)
height of divider in liquid overflow menu

Closes RQA-3646, Closes RQA-3672, Closes RQA-3683

SyntaxColoring committed:
fix(robot-server): Fix `None` ambiguity in labware offset filtering (#17218)

alexjoel42 committed:
feat(protocol-designer): Make createNew.ts more robust and use it to start testing protocol actions in PD  (#17156)

<!--
Thanks for taking the time to open a Pull Request (PR)! Please make sure
you've read the "Opening Pull Requests" section of our Contributing
Guide:


https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

GitHub provides robust markdown to format your PR. Links, diagrams,
pictures, and videos along with text formatting make it possible to
create a rich and informative PR. For more information on GitHub
markdown, see:


https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

To ensure your code is reviewed quickly and thoroughly, please fill out
the sections below to the best of your ability!
-->

# Overview
These additions to the test case of transferSetting.js.cy are the start
of providing greater test coverage to our transfer form.

At this moment, it only goes up to saving a liquid and then opening the
transfer form.

<!--
Describe your PR at a high level. State acceptance criteria and how this
PR fits into other work. Link issues, PRs, and other relevant resources.
-->

## Test Plan and Hands on Testing
These are changes exclusively to cypress. So it will be tested when
folks run their code.
<!--
Describe your testing of the PR. Emphasize testing not reflected in the
code. Attach protocols, logs, screenshots and any other assets that
support your testing.
-->

## Changelog

1. opentrons/protocol-designer/cypress/e2e/transferSettings.cy.js . 
2. opentrons/protocol-designer/cypress/support/commands.ts added some
additional helper functions

Please add special attention to checking if this was the correct way to
add liquid. Saving it took a bit of extra effort. AFAIK

> When you submit a form, the browser sends a POST or GET request to the
action URL specified in the <form> tag or the current page URL if no
action is defined.
Cypress executes the test and observes the page reload, which can
interfere with your test flow if you're not handling it properly.

So I tried to do the following

```Javascript
cy.contains('button', 'Add liquid').click()
                        cy.contains('button', 'Liquid').click()
                        cy.contains('button', 'Define a liquid').click()
                        cy.get('input[name="name"]') // Select the input with name="name"
                            .type('My liquid!')

                        cy.get('div[aria-label="ModalShell_ModalArea"]')
                            .find('form') // Target the form that wraps the button
                            .invoke('submit', (e) => {
                                e.preventDefault(); // Prevent default behavior
                            });
```
<!--
List changes introduced by this PR considering future developers and the
end user. Give careful thought and clear documentation to breaking
changes.
-->

## Review requests

@y3rsh @jerader 

<!--
- What do you need from reviewers to feel confident this PR is ready to
merge?
- Ask questions.
-->

## Risk assessment
Medium. This might take some extra time with developers to ensure that
I'm not breaking existing testing behavior.
<!--
- Indicate the level of attention this PR needs.
- Provide context to guide reviewers.
- Discuss trade-offs, coupling, and side effects.
- Look for the possibility, even if you think it's small, that your
change may affect some other part of the system.
- For instance, changing return tip behavior may also change the
behavior of labware calibration.
- How do your unit tests and on hands on testing mitigate this PR's
risks and the risk of future regressions?
- Especially in high risk PRs, explain how you know your testing is
enough.
-->

jerader committed:
feat(protocol-designer): yaml upload and download artifact version to 4 (#17229)

github actions upload-artifact and download-artifact v3 was deprecated
so this updates to v4.

vegano1 committed:
feat(api, robot-server, shared-data): add FlexStacker module to the hardware controller and robot server. (#17187)

CaseyBatten committed:
include stack limit in evotips

CaseyBatten committed:
update labware definition for new stacking rules

Laura-Danielle committed:
update XYZ coordinates for the evotips opentrons 96 labware

Laura-Danielle committed:
update compatible parent for evotip definitions
